jenkins/ring-install.sh

```diff
@@ -0,0 +1,324 @@
+
+    # We use 2 alternative methods to add the key because the script can also
+    # be used outside of Jenkins context (in a standalone way)
+    if ! gpg --keyserver keys.gnupg.net --recv-keys 5B1943DD; then
+        sudo apt-key add jenkins/cinder-sofs-validate/scality.gpg
+    else
+        gpg -a --export 5B1943DD | sudo apt-key add -
+    fi
+
+    # snmp-mibs-downloader is a dependency. It is only available in Ubuntu multiverse :
+    sudo sed -ri "s/^#\s+(.*multiverse.*)/\1/" /etc/apt/sources.list
+    sudo apt-get update
+}
+
+function _prepare_datadir_on_tmpfs {
+    sudo mkdir -p /scalitytest/disk1
+    # be sure we don't mount multiple times
+    if [[ -z "$(mount -l | grep /scalitytest/disk1)" ]]; then
+        sudo mount -t tmpfs -o size=60g tmpfs /scalitytest/disk1
+    fi
+}
+
+function _prepare_datadir_on_fs {
+    sudo mkdir -p /scalitytest/disk1
+}
+
+function _install_dependencies {
+    sudo apt-get install --yes debconf-utils snmp
+}
+
+function _tune_base_scality_node_config {
+    local conf_has_changed=false
+    if [[ -z "$(egrep '^dirsync=0' /etc/biziod/bizobj.disk1)" ]]; then
+        echo "dirsync=0" | sudo tee -a /etc/biziod/bizobj.disk1
+        conf_has_changed=true
+    fi
+    if [[ -z "$(egrep '^sync=0' /etc/biziod/bizobj.disk1)" ]]; then
+        echo "sync=0" | sudo tee -a /etc/biziod/bizobj.disk1
+        conf_has_changed=true
+    fi
+    if $conf_has_changed; then
+        sudo service scality-node restart
+    fi
+}
+
+function install_base_scality_node {
+    # See http://docs.scality.com/display/R43/Setting+Up+Credentials+for+Ring+4.3
+    cat > /tmp/scality-installer-credentials <<EOF
+{
+   "gui":{
+       "username":"$SUP_ADMIN_LOGIN",
+       "password":"$SUP_ADMIN_PASS"
+   },
+   "internal-management-requests":{
+       "username":"$INTERNAL_MGMT_LOGIN",
+       "password":"$INTERNAL_MGMT_PASS"
+   }
+}
+EOF
+
+    # A full Tempest volume API run needs at least 40G of disk space
+    if [[ $(free -m | awk '/Mem:/ {print $2}') -gt 65536 ]]; then
+        _prepare_datadir_on_tmpfs
+    else
+        _prepare_datadir_on_fs
+    fi
+    sudo touch /scalitytest/disk1/.ok_for_biziod
+
+    _install_dependencies
+
+    # See https://docs.scality.com/display/R43/Install+Nodes+on+Ubuntu#InstallNodesonUbuntu-Configuringthenodes
+    echo "scality-node scality-node/meta-disks string" | sudo debconf-set-selections
+    echo "scality-node scality-node/set-bizobj-on-ssd boolean false" | sudo debconf-set-selections
+    echo "scality-node scality-node/mount-prefix string /scalitytest/disk" | sudo debconf-set-selections
+    echo "scality-node scality-node/name-prefix string node-n" | sudo debconf-set-selections
+    echo "scality-node scality-node/setup-sagentd boolean true" | sudo debconf-set-selections
+    echo "scality-node scality-node/processes-count string 1" | sudo debconf-set-selections
+    echo "scality-node scality-node/chord-ip string $HOST_IP" | sudo debconf-set-selections
```

Comment:
Well there are lots of others : SUP_ADMIN_LOGIN, SUP_ADMIN_PAS, SUP_ADMIN_PASS, etc..
They are used in differents methods, its seems overkill to assert about them all over the place.
Maybe we could just add a comments about the need to define those variables a the top level of the file.